docs/cookbook/create-basic-grid.md

```diff
@@ -0,0 +1,429 @@
+     */
+    protected $em;
+
+    /**
+     * @param \Doctrine\ORM\EntityManagerInterface $em
+     */
+    public function __construct(EntityManagerInterface $em)
+    {
+        $this->em = $em;
+    }
+
+    /**
+     * @param int $salesmanId
+     * @return \Shopsys\ShopBundle\Model\Salesman\Salesman
+     */
+    public function getById(int $salesmanId): Salesman
+    {
+        $salesman = $this->getSalesmanRepository()->find($salesmanId);
+
+        if ($salesman === null) {
+            // you should throw new custom exception here...
+        }
+
+        return $salesman;
+    }
+
+    /**
+     * @return \Doctrine\ORM\EntityRepository
+     */
+    protected function getSalesmanRepository(): EntityRepository
+    {
+        return $this->em->getRepository(Salesman::class);
+    }
+}
+```
+
+#### 6.1.2 Create `SalesmanFacade` and implement `deleteById` method
+```php
+// src\Shopsys\ShopBundle\Model\Salesman\SalesmanFacade.php
+declare(strict_types=1);
+
+namespace Shopsys\ShopBundle\Model\Salesman;
+
+use Doctrine\ORM\EntityManagerInterface;
+
+class SalesmanFacade
+{
+    /**
+     * @var \Doctrine\ORM\EntityManagerInterface
+     */
+    protected $entityManager;
+
+    /**
+     * @var \Shopsys\ShopBundle\Model\Salesman\SalesmanRepository
+     */
+    protected $salesmanRepository;
+
+    /**
+     * @param \Doctrine\ORM\EntityManagerInterface $entityManager
+     * @param \Shopsys\ShopBundle\Model\Salesman\SalesmanRepository $salesmanRepository
+     */
+    public function __construct(EntityManagerInterface $entityManager, SalesmanRepository $salesmanRepository)
+    {
+        $this->entityManager = $entityManager;
+        $this->salesmanRepository = $salesmanRepository;
+    }
+
+    /**
+     * @param int $salesmanId
+     */
+    public function deleteById(int $salesmanId)
+    {
+        $salesman = $this->salesmanRepository->getById($salesmanId);
+
+        $this->entityManager->remove($salesman);
+        $this->entityManager->flush();
+    }
+}
+```
+
+### 6.2 Add delete action to the controller
+In our `SalesmanController`, we add the new action to delete the salesman using `SalesmanFacade`.
+```diff
+// src/Shopsys/ShopBundle/Controller/Admin/SalesmanController
+...
++ use Shopsys\FrameworkBundle\Component\Router\Security\Annotation\CsrfProtection;
++ use Shopsys\ShopBundle\Model\Salesman\SalesmanFacade;
+
+class SalesmanController extends AdminBaseController
+{
++    /**
++     * @var \Shopsys\ShopBundle\Model\Salesman\SalesmanFacade
++     */
++    protected $salesmanFacade;
+
+-    public function __construct(SalesmanGridFactory $salesmanGridFactory)
++    public function __construct(SalesmanGridFactory $salesmanGridFactory, SalesmanFacade $salesmanFacade)
+     {
+         $this->salesmanGridFactory = $salesmanGridFactory;
++        $this->salesmanFacade = $salesmanFacade;
++    }
+
++    /**
++     * @Route("/dashboard/salesmen/delete/{id}", requirements={"id" = "\d+"})
```

Comment:
`/dashboard/` ?

Reply:
try this thing that works also for me
```php

    /**
     * @Route("/salesman/delete/{id}", requirements={"id" = "\d+"})
     * @CsrfProtection
     */
    public function deleteAction(int $id)
    {
        $this->salesmanFacade->deleteById($id);

        $this->getFlashMessageSender()
            ->addInfoFlash(t('Salesman with id %id% deleted', ['%id%' => $id]));

        return $this->redirectToRoute('admin_salesman_list');
    }
```

Reply:
thanks, that was a leftover after the first implementation iteration :slightly_smiling_face: